packages/api/src/repository/library_item.ts

```diff
@@ -25,7 +25,7 @@ export const libraryItemRepository = appDataSource
         .andWhere('md5(original_url) = md5(:url)', { url })
 
       if (forUpdate) {
-        qb.setLock('pessimistic_write')
+        qb.setLock('pessimistic_read')
```

Comment:
I replaced `pessimistic_write` with `pessimistic_read` lock strategy so it will prevent other transactions from modifying the data while it is being read but allows other transactions to read the data concurrently.

Reply:
@jacksonh